docs/current_docs/quickstart/basics/basics.mdx

```diff
@@ -0,0 +1,288 @@
+
+```shell
+container |
+  from alpine
+```
+
+[FIXME recording]
+
+## Adding files and directories
+
+Here's another example, this time adding a file to the container:
+
+```shell
+container |
+  from alpine |
+  with-new-file /hi.txt "Hello from Dagger!"
+```
+
+You could instead add a directory, which could be either a directory from the Dagger host's filesystem or a remote Git repository. Here's an example of adding Dagger's own open source GitHub repository to the container:
+
+```shell
+container |
+  from alpine |
+  with-directory https://github.com/dagger/dagger
+```
+
+## Using the interactive terminal
+
+Did it work? To look inside, request an interactive terminal session with the running container:
+
+```shell
+container |
+  from alpine |
+  with-directory https://github.com/dagger/dagger |
+  terminal
+```
+
+This drops you into an interactive terminal running the `bash` shell. You can now directly execute commands in the running container, as shown below:
+
+[FIXME recording]
+
+:::tip INTERACTIVE CONTAINER DEBUGGING
+The `terminal` function is very useful for debugging and experimenting, since it allows you to interact directly with containers and inspect their state, at any stage of your Dagger Function execution.
+:::
+
+## Chaining
+
+What you just did (and have been doing since the start) is chaining one Dagger API function call to another with the pipe (`|`) operator. This is one of Dagger's most powerful features, as it allows you create dynamic workflows in a single command - no context switching between Dockerfile creation, build commands, and registry pushes.
+
+Our documentation has numerous examples of function chaining in action but here's one more: creating an Alpine container, dropping in a text file with a custom message, setting it to display that message when run, and publishing it to a temporary registry - all in a single command!
+
+```shell
+container |
+  from alpine |
+  with-new-file /hi.txt "Hello from Dagger!" |
+  with-entrypoint cat /hi.txt |
+  publish ttl.sh/hello
+```
+
+[FIXME recording]
+
+## Getting help
+
+The Dagger API is extensively documented but if you're unsure how to proceed at any point, simply append `.help` to your in-progress workflow for context-sensitive assistance on what you can do next:
+
+```shell
+container |
+  from alpine |
+  .help
+```
+
+[FIXME recording]
+
+## Writing custom functions
+
+As your workflows become more complex, you'll start wishing you could make them more reusable, repeatable and shareable. To do this, encapsulate your workflows into custom Dagger Functions. These are just regular code, written in a programming language using a type-safe Dagger SDK.
+
+Here's the previous example, rewritten as a Dagger Function:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./snippets/functions/go/main.go
```

Comment:
I'm going to make my pitch again for just showing partial snippets here and not full code files. It's too much context and someone reading this for the first time doesn't know what the "important" part is. It's also just a bunch of extra lines that don't add value. Nobody needs to copy/paste full files into their editor, they can copy paste partials and figure out where it needs to go.

This part should _just_ be
```go
dag.Container().
		From("alpine:latest").
		WithNewFile("/hi.txt", "Hello from Dagger!").
		WithEntrypoint([]string{"cat", "/hi.txt"}).
		Publish(ctx, "ttl.sh/hello")
```

Reply:
Shouldn't we have the function signature as well, so that the user can see the arguments and return types? 

I've modified to only include partials as you suggested but kept the function signatures, let me know if this is fine or if we should reduce it further.

Reply:
Yeah this works